### DISCUSSION TITLE
Stuck on winget 1.3, what's the best way to get to latest?

### DISCUSSION THREAD

**Discussion by halr9000** [microsoft/winget-cli#3059] [Q&A] [answered]
TL;DR have 1.3, looking at #2421 there's danger here. What is the best way to get back on track to 1.4/latest?

Been getting some winget install bugs lately, went to dig in and start playing with the PowerShell cmdlets when I noticed this:

```
PS C:\Users\hal> winget -v
v1.3.2691
```

So I look for where it's installed, it's under my appdata folder, meaning something put it there which wasn't me.
```
PS C:\Users\hal> gcm winget

CommandType     Name                                               Version    Source
-----------     ----                                               -------    ------
Application     winget.exe                                         0.0.0.0    C:\Users\hal\AppData\Local\Microsoft\Win…
```

Then I look at winget installed and see that it's not 1.4

```
PS C:\Users\hal> winget list winget
Name                                     Id                                    Version           Source
--------------------------------------------------------------------------------------------------------
WingetUI                                 SomePythonThings.WingetUIStore        1.6.2             winget
Windows Package Manager Manifest Creator Microsoft.WingetCreate                1.2.5.0           winget
Windows Package Manager Source (winget)  Microsoft.Winget.Source_8wekyb3d8bbwe 2023.309.1735.199
```

For the heck of it, I went to the package manager app in MS Store. There are no install/update/remove buttons visible. 

Looking at code and #2421, this is/was a legit ID. But it has no upgrade path. The issue doesn't answer the question for me -- how do I get from 1.3 to 1.4?

I'll figure it out...but am multitasking and would love some ideas from y'all.

**Comment by Trenly** [comment, marked as answer]
I would try running this in an admin terminal - 
```
Add-AppxPackage -Path 'https://github.com/microsoft/winget-cli/releases/download/v1.4.10173/Microsoft.DesktopAppInstaller_8wekyb3d8bbwe.msixbundle'
```

**Reply by halr9000** [reply]
Sweet, looks good!

```
# Note: this one failed at first until I killed WindowsPackageManagerServer process.
PS C:\Users\hal> Add-AppxPackage -Path 'https://github.com/microsoft/winget-cli/releases/download/v1.4.10173/Microsoft.DesktopAppInstaller_8wekyb3d8bbwe.msixbundle'

PS C:\Users\hal> winget list winget
Name                                     Id                                    Version           Source
--------------------------------------------------------------------------------------------------------
WingetUI                                 SomePythonThings.WingetUIStore        1.6.2             winget
Windows Package Manager Manifest Creator Microsoft.WingetCreate                1.2.5.0           winget
Windows Package Manager Source (winget)  Microsoft.Winget.Source_8wekyb3d8bbwe 2023.309.2143.126
PS C:\Users\hal> winget -v
v1.4.10173

PS C:\Users\hal> gmo -list PackageManagement

    Directory: C:\program files\powershell\7\Modules

ModuleType Version    PreRelease Name                                PSEdition ExportedCommands
---------- -------    ---------- ----                                --------- ----------------
Script     1.4.8.1               PackageManagement                   Desk      {Find-Package, Get-Package, Get-Package…

```

**Comment by mdanish-kh** [comment]
> For the heck of it, I went to the package manager app in MS Store. There are no install/update/remove buttons visible.

Did you search "App Installer" in MS Store? I want to know if it's the same case as https://github.com/microsoft/winget-pkgs/issues/98936#issuecomment-1460668002

**Reply by halr9000** [reply]
No buttons in there, so it acts as if it's up-to-date. I'll watch that other issue, unsure if related, but it's recent, so.

![image](https://user-images.githubusercontent.com/18482/224153890-7368e951-8bad-49e0-9f14-300d984cf7d4.png)
